common/types/head_listener.go

```diff
@@ -0,0 +1,22 @@
+package types
+
+import "context"
+
+// HeadListener is a chain agnostic interface that manages connection of Client that receives heads from the blockchain node
+type HeadListener[H Head[BLOCK_HASH], BLOCK_HASH Hashable] interface {
+	// ListenForNewHeads kicks off the listen loop (not thread safe)
+	// done() must be executed upon leaving ListenForNewHeads()
+	ListenForNewHeads(handleNewHead NewHeadHandler[H, BLOCK_HASH], done func())
+
+	// ReceivingHeads returns true if the listener is receiving heads (thread safe)
+	ReceivingHeads() bool
+
+	// Connected returns true if the listener is connected (thread safe)
+	Connected() bool
+
+	// HealthReport returns report of errors within HeadListener
+	HealthReport() map[string]error
```

Comment:
It's unusual to see this coupled to anything other than the generic `Service` type. Where is it called from?

Reply:
This is called from headtracker. 

https://github.com/smartcontractkit/chainlink/blob/40e7344f903ea4fdc672ac2ff27aa21c5b954e03/core/chains/evm/headtracker/head_tracker.go#L150

Reply:
But it doesn't need the rest of the `ServiceCtx`? :thinking: 
```go
type ServiceCtx interface {
	Start(context.Context) error
	Close() error
	Name() string
	Ready() error
	HealthReport() map[string]error
}
```
Typically we just embed the `ServiceCtx`:
```suggestion
	services.ServiceCtx
```

Reply:
It looks like HeadListener just requires `HealthReport() map[string]error`. In that case, would it be cleaner to just have `HealthReport() map[string]error`? Or would having services.ServiceCtx be more relevant?

Reply:
I mean that only using `HealthReport()` is a very strange asymmetry. How is it the case that you need to call this method, but you do _not_ need to call `Close()`, for example? Are we missing `Close()` calls? Or are those done against a differently typed variable that we could also use for `HealthReport()`?

This is important because `HealthReport()` is not just some random method - it has very specific implementation requirements, and is a fundamental part of defining a service.

Reply:
This might be out of scope, but there is something wrong with the way this pattern is being used.

Reply:
Looking at the code, i think there is something wrong with the way `HealthReport()` pattern is being used. I can create a ticket and tag in @prashantkumar1982 and @aalu1418 to see how we can continue with this issue :)